vivify/CMakeLists.txt

```diff
@@ -12,25 +12,32 @@
 cmake_minimum_required(VERSION 3.4.1)
 project(vivify C CXX)
 find_package(PkgConfig REQUIRED)
+include(CheckSymbolExists)
+include(CheckIncludeFile)
 
 # vivify external dependencies
 #   needs libunwind, libunwind-x86_64 to extract instruction pointers from stack frames
-#   Unwinding stack needs extraction of eh_frame section, which in turn requires
-#     1. libelf to open and extract eh_frame header and data
-#     2. dwarf.h to read eh_fram with proper exception header encoding
+#   unwinding stack needs extraction of eh_frame section, which in turn requires:
+#     1. libelf to extract eh_frame header
+#     2. dwarf.h to decode eh_frame fields
 #   needs libbfd to map instruction pointers to function, file and line no
-# TODO: make eh_frame extraction optional, if dwarf.h can't be located
 
 check_include_file("dwarf.h" DWARF_H_FOUND)
 if(NOT DWARF_H_FOUND)
-  message(WARNING "skipping vivify build. dwarf.h not found !!!")
+  message(WARNING "skipping vivify build. dwarf.h not found!!!")
   return()
 endif()
 check_include_file("bfd.h" BFD_H_FOUND)
 if(NOT BFD_H_FOUND)
-  message(WARNING "skipping vivify build. bfd.h not found !!!")
+  message(WARNING "skipping vivify build. bfd.h not found!!!")
   return()
 endif()
+find_library(LIBELF_FOUND elf)
```

Comment:
don't we also need unwind-x86_64, unwind, bfd ?

Reply:
Yes, we check those in other places. Also, without libunwind we do not stop because we still can compile/install uitls like mapping, elf, dwarf, bfd.